core/src/test/scala/io/qbeast/core/transform/LinearTransformationTest.scala

```diff
@@ -76,4 +76,29 @@ class LinearTransformationTest extends AnyFlatSpec with Matchers {
 
   }
 
+  it should "merge IdentityTransformations correctly" in {
+    val nullValue = 5000
+    val linearT = LinearTransformation(0, 10000, nullValue, IntegerDataType)
+
+    var otherNullValue = LinearTransformationUtils.generateRandomNumber(0, 90000, Option(42.toLong))
+    linearT.merge(IdentityTransformation(90000)) shouldBe LinearTransformation(
+      0,
+      90000,
+      otherNullValue,
+      IntegerDataType)
+
+    otherNullValue = LinearTransformationUtils.generateRandomNumber(-100, 10000, Option(42.toLong))
+    linearT.merge(IdentityTransformation(-100)) shouldBe LinearTransformation(
+      -100,
+      10000,
+      otherNullValue,
+      IntegerDataType)
+
+    otherNullValue = LinearTransformationUtils.generateRandomNumber(0, 10000, Option(42.toLong))
+    linearT.merge(IdentityTransformation(10)) shouldBe LinearTransformation(
```

Comment:
I think this test is not correct.

Because if an `IdentityTransformation` falls in the `LinearTransformation` min-max, then the existing `LinearTransformation` is used. The `merge` method is called only if `isSuperseededBy` returns true. 

Probably, in this case, we should test how `isSuperseededBy` behaves.

Reply:
You are right, the logic in isSupersededBy is not tested by this. I will add a test in `TransformerTest` for this.